converters/toZigbee.js

```diff
@@ -1154,6 +1154,19 @@ const converters = {
             await entity.read('hvacThermostat', ['ctrlSeqeOfOper']);
         },
     },
+    thermostat_programming_operation_mode: {
+        key: ['programming_operation_mode'],
+        convertSet: async (entity, key, value, meta) => {
+            let val = utils.getKey(constants.thermostatProgrammingOperationModes, value, undefined, Number);
+            if (val === undefined) {
```

Comment:
Why not just
`let val = utils.getKey(constants.thermostatProgrammingOperationModes, value, value, Number);`

There result of that should be the same as far as I can see?

Reply:
Hah, that was a terrible case of blind copy-pasta. It's taken from the "control sequence of operation" thing above it, and the snippet is as broken there as it is here...

I have changed it to throw an exception with the valid options instead.

Reply:
Throwing an exception does indeed make more sense!